test/test.c

```diff
@@ -33969,6 +33968,118 @@ void test_MicroGroups(void)
     _stopServer(serverPid);
 }
 
+void test_MicroQueueGroupForEndpoint(void)
+{
+    microError *err = NULL;
+    natsPid serverPid = NATS_INVALID_PID;
+    natsOptions *opts = NULL;
+    natsConnection *nc = NULL;
+
+    microServiceConfig serviceConfig = { .Name = "testService", .Version = "1.0.0" };
+    microGroupConfig group1Config = { .Prefix = "testGroup1" };
+    microGroupConfig group2Config = { .Prefix = "testGroup2" };
+    microEndpointConfig epConfig = { .Name = "testEP", .Handler = _microHandleRequest42 };
+
+    typedef struct {
+        const char *name;
+
+        bool serviceNoQueueGroup;
+        const char *serviceQueueGroup;
+        bool group1NoQueueGroup;
+        const char *group1QueueGroup;
+        bool group2NoQueueGroup;
+        const char *group2QueueGroup;
+        bool epNoQueueGroup;
+        const char *epQueueGroup;
+
+        const char *expectedServiceLevel;
+        const char *expectedGroup1Level;
+        const char *expectedGroup2Level;
+    } TC;
+    TC tcs[] = {
+        {.name="default",
+            .expectedServiceLevel="q", .expectedGroup1Level="q", .expectedGroup2Level="q"},
```

Comment:
Maybe replace `"q"` with `MICRO_DEFAULT_QUEUE_GROUP` since we include the header that defines it? True for all references of "q" in this file I believe.